Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromCondDB.cc

```diff
@@ -76,65 +79,106 @@ GEMGeometry* GEMGeometryBuilderFromCondDB::build(const RecoIdealGeometry& rgeo)
+      GEMSuperChamber* sch = new GEMSuperChamber( schid, surf );
+      LogDebug("GEMGeometryBuilder") << "GEM SuperChamber created with id = " << schid
+				     << " and added to the GEMGeometry" << std::endl;
+      superChambers.emplace_back( sch );
+    }
+    
     // Create the chamber 
-    GEMChamber* ch = new GEMChamber (chid, surf); 
+    GEMChamber* ch = new GEMChamber( chid, surf ); 
+    LogDebug("GEMGeometryBuilder") << "GEM Chamber created with id = " << chid
+				   << " = " << chid.rawId() << " and added to the GEMGeometry" << std::endl;
+    LogDebug("GEMGeometryBuilder") << "GEM Chamber has following eta partitions associated: " << std::endl;
+
     // Add the etaps to rhe chamber
-    for(std::list<GEMEtaPartition *>::iterator gepl=gepls.begin();
-    gepl!=gepls.end(); gepl++){
-      ch->add(*gepl);
+    for( const auto& gep : gepList ) {
+      ch->add(gep);
+      LogDebug("GEMGeometryBuilder") << "   --> GEM Eta Partition " << GEMDetId(( *gep ).id()) << std::endl;
     }
     // Add the chamber to the geometry
-    geometry->add(ch);
-
+    theGeometry->add( ch );
+  }
+  
+  // The super chamber is composed of 2 chambers.
+  // It's detId is layer 0, chambers are layer 1 and 2
+
+  // construct the regions, stations and rings. 
+  for( int re = -1; re <= 1; re = re+2 ) {
+    GEMRegion* region = new GEMRegion( re );
+    for( int st = 1; st <= GEMDetId::maxStationId; ++st ) {
+      GEMStation* station = new GEMStation(re, st);
+      std::string sign( re==-1 ? "-" : "");
+      std::string name("GE" + sign + std::to_string(st) + "/1");
+      station->setName(name);
+      for( int ri = 1; ri <= 1; ++ri ) {
+	GEMRing* ring = new GEMRing( re, st, ri );
+	for( unsigned sch = 0; sch < superChambers.size(); ++sch ) {
+	  GEMSuperChamber* superChamber = superChambers[sch];
+	  const GEMDetId detId( superChamber->id());
+	  if (detId.region() != re || detId.station() != st || detId.ring() != ri) continue;
+	  
+	  auto ch1 = theGeometry->chamber( GEMDetId( detId.region(), detId.ring(), detId.station(), 1, detId.chamber(), 0 ));
+	  auto ch2 = theGeometry->chamber( GEMDetId( detId.region(), detId.ring(), detId.station(), 2, detId.chamber(), 0 ));
+	  superChamber->add( const_cast<GEMChamber*>( ch1 ));
+	  superChamber->add( const_cast<GEMChamber*>( ch2 ));
+	  ring->add( superChamber );
+	  theGeometry->add( superChamber );
+	  LogDebug("GEMGeometryBuilderFromDDD") << "Adding super chamber " << detId << " to ring: " 
+						<< "re " << re << " st " << st << " ri " << ri << std::endl;
+	}
+	LogDebug("GEMGeometryBuilderFromDDD") << "Adding ring " <<  ri << " to station " << "re " << re << " st " << st << std::endl;
+	station->add( const_cast<GEMRing*>( ring ));
+	theGeometry->add( const_cast<GEMRing*>( ring ));
+      }
+      LogDebug("GEMGeometryBuilderFromDDD") << "Adding station " << st << " to region " << re << std::endl;
+      region->add(const_cast<GEMStation*>(station));
```

Comment:
hi @jshlee @ianna - didn't we already review/discuss the same stanza of code with the same const_cast problems once before?

Reply:
@davidlange6 - yes, we did and it was fixed here:
https://github.com/jshlee/cmssw/commit/d64f6e0e9e1c8366d11b9513dbb59fdd37dd9304#diff-991663c85a84d7e4630761ba3d046958L163